### PR TITLE
fix(flutterfire_ui): use `EmailVerificationScreen.actionCodeSettings` & and fix `flutter analyze` with Flutter 3.0.0

### LINKED ISSUE #8645: [flutterfire_ui] `flutter analyze` fails with Flutter 3.0.0 and breaks the CI
See: https://github.com/firebase/flutterfire/runs/6398175458?check_suite_focus=true

### LINKED ISSUE #8646: [firebase_dynamic_links_platform_interface] `flutter analyze` fails with Flutter 3.0.0
See: https://github.com/firebase/flutterfire/runs/6398175458?check_suite_focus=true

### LINKED ISSUE #8650: 🐛 [flutterfire_ui] Parameter `EmailVerificationScreen.actionCodeSettings` isn't used
Open [email_verification_screen.dart](https://github.com/firebase/flutterfire/blob/master/packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart) and scroll to line 27. You will see that the `actionCodeSettings` isn't passed to the `_EmailVerificationScreenContent` screen:

https://github.com/firebase/flutterfire/blob/374c9df33bbb6b354ea526dcc6cc7812fa4452c0/packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart#L60-L62

### PR DESCRIPTION
**Pull request by nilsreichardt**
## Description
Flutter 3.0.0 caused failing the CI: https://github.com/firebase/flutterfire/runs/6398175458?check_suite_focus=true

```
Nils-MBP-13-M1-3:flutterfire_ui nils$ flutter analyze
Analyzing flutterfire_ui...                                             

   info • A value for optional parameter 'actionCodeSettings' isn't ever given • lib/src/auth/screens/email_verification_screen.dart:78:10 •
          unused_element
warning • The '!' will have no effect because the receiver can't be null • lib/src/auth/screens/email_verification_screen.dart:95:30 •
       unnecessary_non_null_assertion

2 issues found. (ran in 3.4s)
```

This PR fixes the `flutter analyze` command. Additionally, the `unused_element` lint found this bug:
* #8650 

## Related Issues
Closes #8650 
Closes #8645
Part of #8644 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart [hunk 1]
```diff
@@ -74,8 +75,8 @@ class _EmailVerificationScreenContent extends StatefulWidget {
 
   const _EmailVerificationScreenContent({
     Key? key,
-    this.auth,
-    this.actionCodeSettings,
+    required this.auth,
+    required this.actionCodeSettings,
```
Comment (nilsreichardt) [inline packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart:79]: I added the `required` operator to ensure a value (no matter if null or non-null) is passed 👍

**Comment by nilsreichardt** [comment]
cc: @Salakar (part of #8645)

**Review by Salakar — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]

**Comment by aebrahim** [comment]
This breaks the build for dart 2.16.2 (used by flutter 2.10.5) which should still be supported, please revert (e.g. by merging #8713)

**Comment by nilsreichardt** [comment]
@aebrahim What should we do? This PR was necessary to work better with Flutter 3.0.0 (fix the CI and get rid of the null-aware options warnings: https://github.com/flutter/flutter/issues/103561). When you need the fix of this PR, just fork the repo and use the fork until you upgraded to Flutter 3.0.0. If you don't need the fix then just do not upgrade the Firebase packages as long as you are using Flutter 2.10.5.

**Comment by aebrahim** [comment]
Thanks @nilsreichardt :)

**Pull request merged**